src/main/java/spoon/reflect/factory/Factory.java

```diff
@@ -130,6 +131,67 @@
  * Otherwise FactoryImpl is a default implementation.
  */
 public interface Factory {
+	String[] keywords = {
+			"abstract", "continue", "for", "new", "switch",
+			"assert", "default", "if", "package", "synchronized",
+			"do", "goto", "private", "this",
+			"break", "implements", "protected", "throw",
+			"else", "import", "public", "throws",
+			"case", "enum", "instanceof", "return", "transient",
+			"catch", "extends", "try",
+			"final", "interface", "static",
+			"class", "finally", "strictfp", "volatile",
+			"const", "native", "super", "while", "_",
+			"true", "false", "null"
+			// primitive types are removed as they can be used in type reference
```

Comment:
I needed in sniper mode algorithms the list of java keywords too. But I needed complete list. So if this list is not complete then I suggest to make it even private - because it makes sense only for this one algorithm.
WDYT?

Reply:
Another way to do it is to keep a complete list of keywords, and to add the primitive types as the exceptions below. Maybe it would bring also a better semantic. WDYT?

Reply:
yes, sounds good. Note: sniper mode needs also list of operators, separators, ... and some kind of java parser utilities ... I would say all of them (including keywords) should stay internal in spoon. It is probably not target of spoon to publish this java parser information in API